frontend/src/components/Modal/RiderModalInfo.tsx

```diff
@@ -1,112 +1,157 @@
+                return needsArr.reduce(isValidNeed, true);
+              },
+            })}
           />
-          {errors.email && 'enter some needs'}
+          {errors.needs?.type === "validate" && (
+            <p className={styles.error}>
+              Invalid needs. You can enter 'Assistant', 'Crutches', or
+              'Wheelchair'
+            </p>
+          )}
         </div>
-        <div className={cn(styles.gridR3, styles.gridC2)}>
+        <div className={cn(styles.gridR2, styles.gridCBig2)}>
           <Input
             name="address"
             type="text"
             placeholder="Address"
-            ref={register({ required: true })}
+            ref={register({
+              required: true,
+              pattern: /^[a-zA-Z0-9\s,.'-]{3,}$/,
+            })}
           />
-          {errors.email && 'enter an address'}
+          {errors.address && (
+            <p className={styles.error}>Please enter an address</p>
+          )}
         </div>
-        <div className={cn(styles.gridR4, styles.gridC1)}>
-          <Label htmlFor="start">Start Date:</Label>
-          <Input
-            type="date"
-            name="start"
-            ref={register({ required: true })}
-          />
-        </div>
-        <div className={cn(styles.gridR4, styles.gridC2)}>
-          <Label htmlFor="end">End Date:</Label>
-          <Input
-            type="date"
-            name="end"
-            ref={register({ required: true })}
-          />
+        <div className={cn(styles.gridR3, styles.gridCAll)}>
+          <p>Duration</p>
+          <div style={{ display: "flex" }}>
+            <div>
+              <Input
+                type="date"
+                name="joinDate"
+                ref={register({ required: true })}
+                className={styles.riderDate}
+              />
+              {errors.joinDate && (
+                <p className={styles.error}>Please enter a join date</p>
+              )}
+            </div>
+            <p style={{ margin: "0 1rem" }}>to</p>
+            <div>
+              <Input
+                type="date"
+                name="endDate"
+                ref={register({
+                  required: true,
+                  validate: (endDate) => {
+                    const joinDate = getValues("joinDate");
+                    return joinDate < endDate;
+                  },
+                })}
+                className={styles.riderDate}
+              />
+              {errors.end?.type === "required" && (
+                <p className={styles.error}>Please enter an end date</p>
+              )}
+              {errors.end?.type === "validate" && (
+                <p className={styles.error}>Invalid end time</p>
+              )}
+            </div>
+          </div>
         </div>
       </div>
-      <Button type="submit">Add a Rider</Button>
+      <Button type="submit">Add a Student</Button>
```

Comment:
Last minor fix I swear, on the designs and the employee modal (which this is v similar to) the submit button is right aligned. Generally that's better practice than center-aligned because it fits the flow of how you read forms better (according to info 1300). Unless the designers told you otherwise (in which case please ignore me) can you make the submit button right aligned instead of center aligned